src/ralph/backends/base.py

```diff
@@ -0,0 +1,40 @@
+"""Base ralph backend interface."""
+
+from abc import ABC
+from typing import Union
+
+from pydantic import BaseSettings
+
+from ralph.conf import BaseSettingsConfig, core_settings
+
+
+class BaseBackendSettingsConfig(BaseSettingsConfig):
+    """Base backend pydantic settings configuration."""
+
+    env_file = ".env"
+    env_file_encoding = core_settings.LOCALE_ENCODING
+
+
+class BaseBackendSettings(BaseSettings):
+    """Base backend default configuration."""
+
+    class Config(BaseBackendSettingsConfig):
+        """Pydantic Configuration."""
+
+        env_prefix = "RALPH_BACKENDS__"
+
+
+class BaseBackend(ABC):
+    """Base ralph backend class."""
+
+    settings_class = BaseBackendSettings
+    settings: settings_class
```

Comment:
I don't understand why you have to do this ? Why not defining one attribute ?

Reply:
Well spotted) We do not have to do it) However, static-type checkers (e.g. pylance) seem to struggle to infer the type of the settings attribute(
With the generic version, this issue is not present anymore.